SofaKernel/framework/sofa/simulation/Simulation.cpp

```diff
@@ -490,6 +473,25 @@ void Simulation::unload(Node::SPtr root)
     root->execute<DeleteVisitor>(params);
 }
 
+
+///////////////////// Simulation static/singleton API //////////////////////////////////////////////
+/// @warning this singleton has one limitation: it is easy to create several types of
+/// simulations at the same time (e.g. DAGSimulation + TreeSimulation)
+/// but it does not sound like a huge limitation
+/// The (unique) simulation which controls the scene
+Simulation::SPtr theSimulation  {nullptr};
```

Comment:
I know this isn't related to this PR, but I need to say it :-P

I don't think a singleton here is very elegant, and can lead to overuse of singletons and losing track of what is created and when it is destroyed. I believe it should be up to the application to hold its simulation...or simulations (one should have the possibility to create many simulations all independent from each other without having the impression of not following sofa's principles).

In the case of runSofa, it could create the simulation's instance, hold its reference, and delete it once the GUI or main loop ends. But it should not be the place of the kernel (this file) to do that.